odl/deform/linearized.py

```diff
@@ -0,0 +1,342 @@
+        as the input. In the 4:th point, the displacement is non-zero and hence
+        we expect to get the value of the point 0.2 to the left, that is 1.0.
+
+        >>> import odl
+        >>> space = odl.uniform_discr(0, 1, 5)
+        >>> disp_field = space.vector_field_space.element([[0, 0, 0, -0.2, 0]])
+        >>> op = LinDeformFixedDisp(disp_field)
+        >>> template = [0, 0, 1, 0, 0]
+        >>> print(op([0, 0, 1, 0, 0]))
+        [0.0, 0.0, 1.0, 1.0, 0.0]
+
+        The result depends on the chosen interpolation. If we chose 'linear'
+        interpolation and offset the point half the distance between two
+        points, 0.1, we expect to get the mean of the values.
+
+        >>> space = odl.uniform_discr(0, 1, 5, interp='linear')
+        >>> disp_field = space.vector_field_space.element([[0, 0, 0, -0.1, 0]])
+        >>> op = LinDeformFixedDisp(disp_field)
+        >>> template = [0, 0, 1, 0, 0]
+        >>> print(op(template))
+        [0.0, 0.0, 1.0, 0.5, 0.0]
+        """
+        if domain is None:
+            if not isinstance(displacement.space, ProductSpace):
+                raise TypeError('`displacement.space` {!r} not a '
+                                '`ProductSpace`'.format(displacement.space))
+            if not displacement.space.is_power_space:
+                raise TypeError('`displacement.space` {!r} not a power'
+                                'space'.format(displacement.space))
+            if not isinstance(displacement[0].space, DiscreteLp):
+                raise TypeError('`displacement[0].space` {!r} not an '
+                                '`DiscreteLp`'.format(displacement[0]))
+
+            domain = displacement[0].space
+        else:
+            if not isinstance(domain, DiscreteLp):
+                raise TypeError('`displacement[0]` {!r} not an `DiscreteLp`'
+                                ''.format(displacement[0]))
+
+            displacement = domain.vector_field_space.element(displacement)
+
+        Operator.__init__(self, domain, domain, linear=True)
+
+        self.__displacement = displacement
+
+    def _call(self, template, out=None):
+        """Implementation of ``self(template)``.
+
+        Parameters
+        ----------
+        template : `domain` element
+            Given template that is to be deformed by the fixed
+            displacement field.
+        """
+        return _linear_deform(template, self.__displacement, out)
+
+    @property
+    def adjoint(self):
+        """Adjoint of the operator.
+
+        Returns
+        -------
+        adjoint: `Operator`
+            The adjoint of the operator.
+        """
+
+        # TODO allow users to select what method to use here.
+        div_op = Divergence(domain=self.__displacement.space, method='forward',
+                            padding_method='symmetric')
+        jacobian_det = self.domain.element(
+            np.exp(-div_op(self.__displacement)))
```

Comment:
That reference uses such an approximation, but the exponential term does not (at least not obviously) come from the approximation of `D(id + v)^(-1)`.
In fact, as I read it, the inverse deformation itself is simply approximated as `x --> x - v(x)` in the argument of the template. The exponential only plays a role for the Jacobian, but I don't see how. They write something about a "Lie-Trotter splitting", but to my understanding, this requires that the diffeo `psi^(-1)` already is of the form `exp(-eps*v)`, and only then it follows that the Jacobian can be approximated as `exp(-eps*div(v))`, not in general.

I'm a bit doubtful that we're doing something reasonable at all here, a clear reference or a verification would be very useful.